Server/App/model/deck.py

```diff
@@ -0,0 +1,127 @@
+from utils.serializer import serializer
+
+
+class Deck(object):
+    def __init__(self, uuid, connector=None):
+        self.uuid = uuid
+        self.connector = connector or DatabaseConnector()
+
+    @staticmethod
+    def create(name, user_id, device, tags, cards, connector):
+        connector = connector or DatabaseConnector()
+        deck_id = generate_uuid()
+        try:
+            connector.begin_transaction()
+            connector.call_procedure('CREATE_DECK', deck_id, name, user_id, device, ','.join(tags))
+            for index in xrange(len(cards)):
+                Card.create(deck_id, cards[index]['front'], cards[index]['back'], index,
+                            reposition=False, connector=connector)
+            connector.end_transaction()
+
+        except:
+            connector.abort_transaction()
+            raise
+
+        return Deck(deck_id)
+
+    @staticmethod
+    def get_by_share_code(share_code, connector=None):
+        if share_code is None:
+            return None
+
+        connector = connector or DatabaseConnector()
+        results = connector.query('SELECT uuid FROM Deck WHERE share_code=%s', share_code)
+
+        if len(results) != 1:
+            return None
+        return Deck(results[0]['uuid'])
+
+    def set_name(self, name):
+        self.connector.query_transactionally('UPDATE Deck SET name=%s WHERE uuid=%s', name, self.uuid)
+
+    def set_public(self, public):
+        self.connector.query_transactionally(
+            'UPDATE Deck SET public=%s WHERE uuid=%s', public, self.uuid)
+
+    def set_tags(self, tags):
+        self.connector.call_procedure_transactionally(
+            'SET_DELIMITED_TAGS', self.uuid, ','.join(tags))
+
+    def update_deck_version(self):
+        self.connector.call_procedure_transactionally('INCREMENT_DECK_VERSION', self.uuid)
+
+    def update_user_version(self, user_id, device):
+        self.connector.call_procedure_transactionally(
+            'INCREMENT_USER_DATA_VERSION', user_id, self.uuid, device)
+
+    def rate(self, user_id, rating):
+        self.connector.begin_transaction()
+        self.connector.call_procedure('UNRATE_DECK', user_id, self.uuid)
+        self.connector.call_procedure('RATE_DECK', user_id, self.uuid, rating)
+        self.connector.end_transaction()
+
+    def get_share_code(self):
+        result = self.connector.call_procedure('GET_SHARE_CODE', self.uuid)
+        if len(result) != 1:
+            return None
+
+        if result[0]['share_code'] is not None:
+            return result[0]['share_code']
+
+        # Need to generate a share code.
+        self.connector.begin_transaction()
+        while True:
+            share_code = ''.join(random.choice(string.ascii_uppercase + string.digits)
+                                 for _ in range(SecurityConfig.SHARE_CODE_LENGTH))
+
+            result = self.connector.call_procedure('SET_SHARE_CODE', self.uuid, share_code)
+            if len(result) != 1:
```

Comment:
Looks like this If will stop the loop if the generated share code was a duplicate, but we want to regenerate a new share code and attempt to add it.